argilla-server/src/argilla_server/services/schemas.py

```diff
@@ -0,0 +1,387 @@
+        if cache_key in self._schema_cache:
+            del self._schema_cache[cache_key]
+
+        return workspace
+
+    async def sync_s3_schema_to_metadata(
+        self, db: AsyncSession, workspace: Workspace, schema_name: str, prefix: str = DEFAULT_SCHEMA_S3_PATH
+    ) -> Dict[str, Any]:
+        """
+        Sync a specific schema from S3 to workspace metadata.
+
+        Args:
+            db: Database session
+            workspace: The workspace to sync schemas for
+            schema_name: Name of the schema to sync
+            prefix: S3 prefix to search for schemas
+
+        Returns:
+            Updated schema configuration
+        """
+        if not PANDERA_AVAILABLE:
+            raise RuntimeError("Pandera is not available - schema operations require the extralit package")
+
+        bucket_name = workspace.name
+        schema_path = f"{prefix.rstrip('/')}/{schema_name}.json"
+
+        try:
+            # Get schema file from S3
+            file_response = files.get_object(self.minio_client, bucket_name, schema_path, include_versions=True)
+
+            # Parse schema to validate it
+            schema_data = json.loads(file_response.response.data.decode("utf-8"))
+            from_json(BytesIO(json.dumps(schema_data).encode("utf-8")))
+
+            # Get current schema configuration
+            current_config = await self.get_workspace_schema_configuration(workspace)
+            schemas = current_config.get("schemas", [])
+
+            # Check if schema is singleton
+            is_singleton = any(
+                check.get("name") == "singleton" and check.get("statistics", {}).get("enabled", True)
+                for check in schema_data.get("checks", [])
+            )
+
+            # Find existing schema or create new entry
+            existing_schema = None
+            for i, s in enumerate(schemas):
+                if s["name"] == schema_name:
+                    existing_schema = i
+                    break
+
+            schema_entry = {
+                "name": schema_name,
+                "is_singleton": is_singleton,
+                "s3_path": schema_path,
+                "version_id": file_response.metadata.version_id,
+                "dependencies": [],  # TODO: Extract dependencies from schema
+            }
+
+            if existing_schema is not None:
+                schemas[existing_schema] = schema_entry
+            else:
+                schemas.append(schema_entry)
+
+            # Update workspace metadata
+            updated_config = {"schemas": schemas, "last_sync": datetime.utcnow().isoformat()}
+
+            await self.update_workspace_schema_configuration(db, workspace, updated_config)
+
+            return updated_config
+
+        except Exception as e:
+            _LOGGER.error(f"Error syncing schema {schema_name} from S3 for workspace {workspace.name}: {e}")
+            raise
+
+    async def sync_all_s3_schemas_to_metadata(
```

Comment:
There are no unit tests covering the sync_s3_schema_to_metadata, sync_all_s3_schemas_to_metadata, or create_schema_structure_from_metadata methods. Adding tests for these will help ensure schema syncing and structure creation behave as expected.